x/launch/types/errors.go

```diff
@@ -11,24 +11,23 @@ var (
-	ErrNotTriggeredLaunch    = sdkerrors.Register(ModuleName, 12, "the chain launch has not been triggered")
-	ErrRevertDelayNotReached = sdkerrors.Register(ModuleName, 13, "the revert delay has not been reached")
-	ErrRequestNotFound       = sdkerrors.Register(ModuleName, 14, "request not found")
-	ErrInvalidConsPubKey     = sdkerrors.Register(ModuleName, 15, "the consensus public key is invalid")
-	ErrInvalidGenTx          = sdkerrors.Register(ModuleName, 16, "the gentx is invalid")
-	ErrInvalidSelfDelegation = sdkerrors.Register(ModuleName, 17, "the self delegation is invalid")
-	ErrInvalidPeer           = sdkerrors.Register(ModuleName, 18, "the peer is invalid")
-	ErrAccountAlreadyExist   = sdkerrors.Register(ModuleName, 19, "account already exists")
-	ErrAccountNotFound       = sdkerrors.Register(ModuleName, 20, "account not found")
-	ErrValidatorAlreadyExist = sdkerrors.Register(ModuleName, 21, "validator already exists")
-	ErrValidatorNotFound     = sdkerrors.Register(ModuleName, 22, "validator not found")
-	ErrChainInactive         = sdkerrors.Register(ModuleName, 23, "the chain is inactive")
+	ErrTriggeredLaunch       = sdkerrors.Register(ModuleName, 4, "launch is triggered for the chain")
+	ErrNoAddressPermission   = sdkerrors.Register(ModuleName, 5, "you must be the coordinator or address owner to perform this action")
+	ErrInvalidCoins          = sdkerrors.Register(ModuleName, 6, "the coin list is invalid")
+	ErrInvalidVestingOption  = sdkerrors.Register(ModuleName, 7, "invalid vesting option")
+	ErrLaunchTimeTooLow      = sdkerrors.Register(ModuleName, 8, "the remaining time is below authorized launch time")
+	ErrLaunchTimeTooHigh     = sdkerrors.Register(ModuleName, 9, "the remaining time is above authorized launch time")
+	ErrNotTriggeredLaunch    = sdkerrors.Register(ModuleName, 10, "the chain launch has not been triggered")
+	ErrRevertDelayNotReached = sdkerrors.Register(ModuleName, 11, "the revert delay has not been reached")
+	ErrRequestNotFound       = sdkerrors.Register(ModuleName, 12, "request not found")
+	ErrInvalidConsPubKey     = sdkerrors.Register(ModuleName, 13, "the consensus public key is invalid")
+	ErrInvalidGenTx          = sdkerrors.Register(ModuleName, 14, "the gentx is invalid")
+	ErrInvalidSelfDelegation = sdkerrors.Register(ModuleName, 15, "the self delegation is invalid")
+	ErrInvalidPeer           = sdkerrors.Register(ModuleName, 16, "the peer is invalid")
+	ErrAccountAlreadyExist   = sdkerrors.Register(ModuleName, 17, "account already exists")
+	ErrAccountNotFound       = sdkerrors.Register(ModuleName, 18, "account not found")
+	ErrValidatorAlreadyExist = sdkerrors.Register(ModuleName, 19, "validator already exists")
+	ErrValidatorNotFound     = sdkerrors.Register(ModuleName, 20, "validator not found")
+	ErrChainInactive         = sdkerrors.Register(ModuleName, 21, "the chain is inactive")
+	ErrChainIsMainnet        = sdkerrors.Register(ModuleName, 22, "the chain is already in mainnet")
```

Comment:
`"the chain is already in mainnet"` the chain is either or not a mainnet, it doesn't become it

Refering to the command above we could maybe change this error type to precise that a mainnet cannot holds account in `launch`